### PR TITLE
Feature/issue 10

### LINKED ISSUE #10: Edit user profile

### PR DESCRIPTION
**Pull request by Faustiks**
# Description

add user_info controller and model with validations 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kapatsila-mariia — APPROVED** [review]

File: server/app/controllers/user_infos_controller.rb [hunk 1]
```diff
@@ -1,5 +1,62 @@
-# frozen_string_literal: true
-
 class UserInfosController < ApplicationController
-  def index; end
+  before_action :authenticate_user!,
+                only: [:index, :update, :create]
+
+  def index
+    show_user_info = current_user.user_info
+    render json: show_user_info.to_json(:only => [ :id, :first_name,
```
Comment (bearded) [inline server/app/controllers/user_infos_controller.rb]: would be good to use json serializer, or separate method that would return json
to follow principle DRY (Don't repeat yourself)

File: server/app/controllers/user_infos_controller.rb [hunk 2]
```diff
@@ -1,5 +1,62 @@
...
+        render json: user_inf, status: 200
+      else
+        render json: user_inf.errors.full_messages, status: :unprocessable_entity
+      end
+    else
+      render json: { message: "U have no rights or info doesn't exist" }, status: 403
+    end
+  end
+
+  def create_user_info_params
```
Comment (bearded) [inline server/app/controllers/user_infos_controller.rb]: seems like one method `user_info_params` would be enough, to not duplicate the same code...

File: server/config/routes.rb [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
-  # devise_for :users
   devise_for :users,
              controllers: {
                sessions: "users/sessions",
                registrations: "users/registrations"
              }
+  get 'user_infos', to: 'user_infos#index'
+  get 'user_infos/:id(.:format)', to: 'user_infos#show'
+  post 'user_infos', to: 'user_infos#create'
+  put 'user_infos', to: 'user_infos#update'
```
Comment (bearded) [inline server/config/routes.rb]: would be good to use singular form of "resource" for UserInfo
e.g.
```ruby
resources :users do
  resource :user_info
end
```
ref: https://guides.rubyonrails.org/routing.html#singular-resources

**Review by bearded — CHANGES_REQUESTED** [review]

**Review by bearded — APPROVED** [review]
Looks good

**Pull request merged**